backend/app/Actions/User/RegisterUserAction.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\Actions\User;
+
+use App\Repositories\UserRepository;
```

Comment:
you bonded interface on implementation, that means you can use here contract.  App\Repositories\Contracts\UserRepository